cli/src/cli.ts

```diff
@@ -161,11 +161,17 @@ export class CLI {
 				this.sessionService.setWorkspaceDirectory(workspace)
 				logs.debug("SessionManager workspace directory set", "CLI", { workspace })
 
+				// Inject configuration BEFORE session restoration to ensure the extension
+				// has the correct config (including org ID) during session operations
+				if (this.options.session || this.options.fork) {
+					await this.injectConfigurationToExtension()
+					logs.debug("CLI configuration pre-injected before session operation", "CLI")
+				}
+
```

Comment:
Kind of hidden here but we do still also injectConfigurationToExtension() again down below on line 184 after the:

```
await this.store.set(loadHistoryAtom)
logs.debug("Command history loaded", "CLI")
```

@iscekic does session impact the config we need to inject ?

Reply:
I don't think it's needed, that atom is used for up/down arrow history.